export/securedrop_export/print/print_dialog.py

```diff
@@ -0,0 +1,95 @@
+import logging
+
+import gi
+
+gi.require_version("Gtk", "4.0")
+
+from gi.repository import Gtk  # noqa: E402
+
+logger = logging.getLogger(__name__)
+
+
+def open_print_dialog(file_to_print):
+    app = PrintDialog(file_to_print)
+    app.run()
+
+
+class PrintDialog(Gtk.Application):
+    def __init__(self, file_to_print):
+        super().__init__(application_id="org.securedrop.PrintDialog")
+        self.file_to_print = file_to_print
+        self.connect("activate", self.on_activate)
+
+    def on_activate(self, app):
+        """On GUI startup"""
+        window = Gtk.Window(application=app)
+        self.dialog = Gtk.PrintUnixDialog.new("Print Document", window)
+        self.dialog.connect("response", self.on_response)
+        self.dialog.show()
+        window.hide()
+
+    def on_response(self, parent_widget, response_id):
+        """
+        When the user clicks one of the print dialog's action buttons
+        [ CANCEL ] [ PRINT ]
+        """
+
+        if response_id == Gtk.ResponseType.OK:  # Print
+            settings = self.dialog.get_settings()
+            printer = self.dialog.get_selected_printer()
+            page_setup = self.dialog.get_page_setup()
+
+            # Fix upstream GTK page ranges:
```

Comment:
How about:

GTK in bookworm has a bug in which page ranges are not properly passed to IPP printers, we work around that by explicitly passing them to CUPS, and disallowing multiple ranges.

And then include links to the two bugs you filed.